### PR TITLE
CI: SENSEI

### PR DESCRIPTION
**Pull request by ax3l**
Add SENSEI to CI.
Adopted from @kwryankrattiger's https://github.com/AMReX-Codes/amrex/pull/2115

- [x] rebase after #2209
- [x] fix the requirement of passing the CXX standard flag (upstream in AMReX) #2236
- [x] rebase after #2265

### CODE REVIEW

File: .github/workflows/insitu.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    container:
+      image: ryankrattiger/sensei:fedora33-vtk-mpi-20210616
+    steps:
+    - uses: actions/checkout@v2
+    - name: Configure
+      run: |
+        cmake -S . -B build     \
+          -DWarpX_SENSEI=ON     \
+          -DWarpX_COMPUTE=NOACC \
+          -DCMAKE_CXX_STANDARD=14
```
Comment (ax3l) [inline .github/workflows/insitu.yml:29]: @c-wetterer-nelson @kwryankrattiger both here and in AMReX we need this flag to compile.
I think that's an upstream issue in AMReX that we should fix, we use `target_compile_features(... PUBLIC cxx_std_14)` for both the upstream `amrex` target as well as WarpX and thus this should not be needed, ideally.

I think there might be a requirement missing of the same kind in AMReX that should be fixed:
https://github.com/AMReX-Codes/amrex/blob/989d98b849ebaf31f000f60ff48ffd09304ff621/.github/workflows/linux.yml#L128

File: .github/workflows/insitu.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  cancel-in-progress: true
+
+jobs:
+  sensei:
+    name: SENSEI
+    runs-on: ubuntu-20.04
+    env:
+      CXX: clang++
+      CC: clang
+      CXXFLAGS: "-Werror -Wshadow -Woverloaded-virtual -Wunreachable-code -fno-operator-names"
```
Comment (ax3l) [inline .github/workflows/insitu.yml]: During rebase, add:
```suggestion
      CXXFLAGS: "-Werror -Wshadow -Woverloaded-virtual -Wunreachable-code -Wno-error=pass-failed"
```

File: .github/workflows/insitu.yml (hunk 1 above)
Comment (ax3l) [reply .github/workflows/insitu.yml:29]: X-ref upstream: https://github.com/AMReX-Codes/amrex/pull/2115#discussion_r691544975

**Comment by kwryankrattiger** [comment]
As you add more CI that uses this container for SENSEI, can you add the PRs that add them to this issue:
https://github.com/SENSEI-insitu/SENSEI/issues/14

File: .github/workflows/insitu.yml (hunk 1 above)
Comment (ax3l) [reply .github/workflows/insitu.yml:29]: X-ref SENSEI: https://github.com/SENSEI-insitu/SENSEI/issues/15

**Comment by ax3l** [comment]
@c-wetterer-nelson cool, now only a few warnings remain of the kind `-Wunused-parameter` :)

We have an `amrex::ignore(..., ..., ...);` for that or you can comment the argument name.
Some of them might also indicate a good point to runtime-verify an assumption made in SENSEI functions.

**Comment by ax3l** [comment]
@c-wetterer-nelson do you like to fix those warnings in a PR? Afterwards I would merge this in to establish SENSEI CI.

**Comment by c-wetterer-nelson** [comment]
> @c-wetterer-nelson do you like to fix those warnings in a PR? Afterwards I would merge this in to establish SENSEI CI.

https://github.com/ECP-WarpX/WarpX/pull/2265 Cheers!

**Comment by ax3l** [comment]
Thanks a lot! Will rebase this PR as soon as we have that one failing test since earlier this morning figured out... (#2269)

**Comment by ax3l** [comment]
@c-wetterer-nelson  can you please rebase this PR? :)

**Comment by ax3l** [comment]
oh lol, it's mine

**Comment by ax3l** [comment]
@RemiLehe CI for SENSEI: ready for review & merge :)

**Comment by ax3l** [comment]
I will rebase this once more - even though it passes it is now 2+ weeks open on green and might not pass after merge

**Review by EZoni — APPROVED** [review]
Thank you, Axel! Will enable auto-merge on this one.

**Pull request merged**